### PR TITLE
Support TIMED=1 in test suite + some preliminary cleanups

### LINKED ISSUE #17943: CI should display timing info for the test-suite
I'm not sure how to best make this happen: something in the external script that prints timing info on every line?  Adding some makefile magic to the test-suite makefile time coqc?

### PR DESCRIPTION
**Pull request by SkySkimmer**
coqwc, coqdoc and tools tests don't use it due to low
expected value but would be trivial to add.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run ci

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: test-suite/Makefile
Comment (SkySkimmer) [inline test-suite/Makefile:207]: @ejgallego what's the right way to get the path of make-one-time-file in dune?
It's supposed to be installed somewhere in coq-core

Comment (ejgallego) [reply test-suite/Makefile:207]: You want to get the path to the build tree or the install tree?

Is this file decalred as a dependency of the rule?

Comment (ejgallego) [reply test-suite/Makefile:207]: Indeed, it is not, that's the problem.

Comment (SkySkimmer) [reply test-suite/Makefile:207]: It's included as part of coq-core in the install tree
(it's used by coq_makefile / timing tests)

Comment (ejgallego) [reply test-suite/Makefile:207]: IMHO add the dependency to dune so the file is pulled from the source tree instead of the installed tree.

The source tree is here anyways.

Comment (ejgallego) [reply test-suite/Makefile:207]: There are few libraries that allow you to query for installed files, but IMHO we don't need to be so fancy here.

File: test-suite/dune
Comment (SkySkimmer) [inline test-suite/dune:19]: all bench machines are busy so will test tomorrow I guess

**Comment by SkySkimmer** [comment]
Looks like it works. The timings are in https://coq.gitlabpages.inria.fr/-/coq/-/jobs/3600666/artifacts/_bench/opam.NEW/ocaml-NEW/.opam-switch/build/coq.dev/_build/default/test-suite/time-of-build.log (no OLD for this run since it's against master)
It seems dune ate the output so they're not copied in _bench/logs/coq.NEW.opam_install.1.stdout.log and so not included in ALL.BOTH.perfile_timings.1.log etc

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
ping @coq/test-suite-maintainers

**Comment by herbelin** [comment]
Timing, async, and yml is not a part of the test suite that I'm familiar with, but I'll merge if I the third test-suite maintainer has no more comments to add.

**Comment by JasonGross** [comment]
(sorry @ejgallego I didn't mean to rerequest a review)

**Comment by SkySkimmer** [comment]
ping @coq/test-suite-maintainers

**Comment by JasonGross** [comment]
Do you want to also add support for `TIMING=1` while you're at it, so we can get per-line diffs?

**Comment by SkySkimmer** [comment]
Maybe someday but not in this PR

**Comment by SkySkimmer** [comment]
@coq/test-suite-maintainers can this get an assignee (and merge?)?

**Comment by herbelin** [comment]
If @JasonGross and @ejgallego do not object, I'll merge tomorrow (I'm not familiar myself with timing).

**Comment by JasonGross** [comment]
I don't have time to review right now, but the implementation looked okay at a quick glance, and I strongly support this change

**Comment by ejgallego** [comment]
I don't objec to this; however I think we should stop updating `test-suite/Makefile` and focus / decide on a an alternative system. The more things we add to `test-suite/Makefile` the more we waste time doubly, on the addition to it, and having to port to whatever next systems succeeds the current gargantuan Makefile (so in a sense, this PR add technical debt)

As of today the experience of running the test suite is very very poor, to the point that a few people have been discouraged of even trying to contribute to Coq when they had to face it.

The main leading alternative is Ali's work on porting the test-suite to Dune; that work is IMHO netly superior to what we have now, and adding features like the ones in this PR tends to be both easy and maintenable (note that I didn't try to look at this concrete PR, so here I'm just guessing)

**Comment by herbelin** [comment]
I then propose to postpone to Monday when we'll be able to talk live.

**Comment by SkySkimmer** [comment]
I think this should still be merged even if we plan to eventually replace the makefile. It is not that complex a change IMO, and anyway it's for devs not users so if the port requires dropping the feature we can do it (unless the feature is considered important, but if it's important then we should merge).

**Comment by herbelin** [comment]
It looks like a good argument. Emilio, I propose to merge which will not mean that it cannot change in the future.

**Comment by ejgallego** [comment]
Sure, as noted I don't oppose this PR at all, ln the other hand I'm afraid I can't help a lot with the make side.

**Review by herbelin — APPROVED** [review]
See discussion

**Comment by herbelin** [comment]
OK. @coqbot merge now

**Comment by herbelin** [comment]
@coqbot merge now

**Pull request merged**
